processor.go

```diff
@@ -136,19 +150,51 @@ func (p *ServiceProcessor) ProcessClientRequest(req *http.Request, path string,
 		arg.Elem().Set(reflect.ValueOf(msg).Elem())
 		ret := f.Call([]reflect.Value{arg})
 
+		// there will be an extra element in ret
 		ierr := ret[1].Interface()
 		if ierr != nil {
 			return nil, ierr.(error)
 		}
 		return ret[0].Interface(), nil
 	}()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
+
+	if reply != nil && reflect.TypeOf(reply).Kind() == reflect.Chan {
```

Comment:
This would be cleaner if we had different handlers for simple requests and streams.